SafeExamBrowser.UserInterface.Contracts/Browser/IBrowserControl.cs

```diff
@@ -56,6 +56,11 @@ public interface IBrowserControl
 		/// </summary>
 		void Destroy();
 
+		/// <summary>
+		/// Executes the given Javascript code in the browser.
+		/// </summary>
+		void ExecuteJavascript(string javascript, System.Action<dynamic> callback);
```

Comment:
It is required to interact with the webpage that is currently loaded in the CEF browser control. Currently I use it in two scenarios:

* When the user uses the tab key to go through the focusable elements of the webpage, I need to detect when they reach the end of the page, to be able to take away the focus from the webpage control. In this case, `ExecuteJavascript` returns a string from Javascript.
* When the browser control receives focus because the user presses tab while e.g. the Menu button has focus, I need to execute some Javascript to make the webpage select the first (or last) focusable element in the webpage. This does not return anything.

If you see other options to do this, then we can gladly remove the `ExecuteJavascript` method.

You're right, the parameter is of type `CefSharp.JavascriptResponse`. Can I use `CefSharp` types in the `SafeExamBrowser.UserInterface.Contracts.Browser` assembly? If yes, this would require a reference to the `CefSharp` assembly. If no, I'm going to create a copy of the `JavascriptResponse` type.

Reply:
Inside of `JavascriptResponse` there is `object Result`, which can be any Javascript object. In my use cases, it is always a string, so we could limit it to a string, but this would limit future uses of `ExecuteJavascript`.

Reply:
Aha, I see. Okay, that makes sense. I guess we could use an `object` then as well (instead of `dynamic`), or is there a reason why you'd prefer `dynamic`?

Reply:
If the thing that comes back is a Javascript object, I would prefer `dynamic` over `object` since it allows me to directly access the properties I expect. For example, when returning

```
return {
  foo: 1,
  bar: "yo"
}
```

in Javascript, I could just access that with a C# dynamic:

```
int foo = javascriptResult.Result.foo;
```

With an `object`, this is much more cumbersome.

But since I do not return objects from Javascript, I don't object to using `object` 😊

Can you please answer my other question: Can I use `CefSharp` types in the `SafeExamBrowser.UserInterface.Contracts.Browser` assembly? If yes, this would require a reference to the `CefSharp` assembly. If no, I'm going to create a copy of the `JavascriptResponse` type.

Reply:
I see, indeed, that makes sense. No, leave it as it is then, that appears to be an appropriate usage of `dynamic` I'd reckon. Thanks for the clarification!

Reply:
I just added `JavascriptResult.cs`, a class with the following members:

```
public bool Success { get; set; }
public string Message { get; set; }
public dynamic Result { get; set; }
```

This way, only `Result` is a `dynamic`. `Success` and `Message` can be strongly typed.